pywr/parameters/_polynomial.pyx

```diff
@@ -0,0 +1,140 @@
+""" Module contains Polynomial Parameters """
+from ._parameters import load_parameter
+import numpy as np
+cimport numpy as np
+
+
+cdef class Polynomial1DParameter(Parameter):
+    """ Parameter that returns the result of 1D polynomial evaluation
+
+    The input to the polynomial can be either:
+     - The previous flow of the attached node (default)
+     - The previous flow of another `AbstractNode`
+     - The current storage of an `AbstractStorage` node
+     - The current value of another `Parameter`
+
+    Parameters
+    ----------
+    coefficients : array like
+        The 1 dimensional array of polynomial coefficients.
+    node : `AbstractNode`
+        An optional `AbstractNode` the flow of which is used to evaluate the polynomial.
+    storage_node : `Storage`
+        An optional `Storage` node the volume of which is used to evaluate the polynomial.
+    parameter : iterable of Parameter objects or single Parameter
+        An optional `Parameter` the value of which is used to evaluate the polynomial.
+    use_proportional_volume : bool
+        An optional boolean only used with a `Storage` node to switch between using absolute
+         or proportional volume when evaluating the polynomial.
+    """
+    def __init__(self, coefficients, *args, **kwargs):
+        self.coefficients = np.array(coefficients, dtype=np.float64)
+        self._other_node = kwargs.pop('node', None)
+        self._storage_node = kwargs.pop('storage_node', None)
+        self._parameter = kwargs.pop('parameter', None)
```

Comment:
The parameter should be added to this parameter's children. `self.children.add(parameter)`